mobify/platform-scaffold
# Web 1555  sfcc address typecheck

This PR fixes some bugs on the checkout shipping page for users with saved addresses.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1555
 **Linked PRs**: https://github.com/mobify/platform-scaffold/pull/1042

## Changes
- Clear shipping address initial values when user selects "Add New Address"
- Prevent JS error when splitting full name and full name is undefined

## Todos

## How to test-drive this PR
- Run `npm start`
- Preview [SFCC](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- log into an account with saved addresses
- add an item to the cart
- proceed to the checkout shipping page
- select "Add new address"
- the address form should be empty
- switch back to a saved address
- try proceeding to the payment page
- you should be able to proceed to the next page
